packages/hub/node-tests/routes/push-notification-registrations-test.ts

```diff
@@ -0,0 +1,200 @@
+      pushClientId: 'FIREBASE_USER_ID',
+      disabledAt: '2021-12-01 10:00:00',
+    });
+
+    let payload = {
+      data: {
+        type: 'push-notification-registration',
+        attributes: {
+          'push-client-id': 'FIREBASE_USER_ID',
+        },
+      },
+    };
+
+    await request()
+      .post('/api/push-notification-registrations')
+      .send(payload)
+      .set('Authorization', 'Bearer abc123--def456--ghi789')
+      .set('Accept', 'application/vnd.api+json')
+      .set('Content-Type', 'application/vnd.api+json')
+      .expect(201)
+      .expect(function (res) {
+        res.body.data.id = 'id';
+      })
+      .expect({
+        data: {
+          id: 'id',
+          type: 'push-notification-registration',
+          attributes: {
+            'owner-address': stubUserAddress,
+            'push-client-id': 'FIREBASE_USER_ID',
+            'disabled-at': null,
+          },
+        },
+      })
+      .expect('Content-Type', 'application/vnd.api+json');
+
+    let records = await pushNotificationRegistrationQueries.query({
+      ownerAddress: stubUserAddress,
+      pushClientId: 'FIREBASE_USER_ID',
+    });
+
+    expect(records.length).to.equal(1);
+    expect(records[0].ownerAddress).to.equal(stubUserAddress);
+    expect(records[0].pushClientId).to.equal('FIREBASE_USER_ID');
+  });
+});
+
+describe('DELETE /api/push-notification-registrations', function () {
+  this.beforeEach(function () {
+    registry(this).register('authentication-utils', StubAuthenticationUtils);
+  });
+  let { request, getContainer } = setupHub(this);
+
+  it('returns 401 without bearer token', async function () {
+    await request()
+      .post('/api/push-notification-registrations')
+      .send({})
+      .set('Accept', 'application/vnd.api+json')
+      .set('Content-Type', 'application/vnd.api+json')
+      .expect(401)
+      .expect({
+        errors: [
+          {
+            status: '401',
+            title: 'No valid auth token',
+          },
+        ],
+      })
+      .expect('Content-Type', 'application/vnd.api+json');
+  });
+
+  it('deletes push notification registration', async function () {
+    let pushNotificationRegistrationQueries = await getContainer().lookup('push-notification-registration-queries');
+
+    await pushNotificationRegistrationQueries.insert({
+      id: shortUuid.uuid(),
+      ownerAddress: stubUserAddress,
+      pushClientId: 'FIREBASE_USER_ID',
+      disabledAt: null,
+    });
+
+    await request()
+      .delete(`/api/push-notification-registrations/FIREBASE_USER_ID`)
+      .send({})
+      .set('Authorization', 'Bearer abc123--def456--ghi789')
+      .set('Accept', 'application/vnd.api+json')
+      .set('Content-Type', 'application/vnd.api+json')
+      .expect(200);
+
+    let records = await pushNotificationRegistrationQueries.query({
+      ownerAddress: stubUserAddress,
+      pushClientId: 'FIREBASE_USER_ID',
+    });
+
+    expect(records.length).to.equal(0);
```

Comment:
I wonder if it would be better or worse to soft delete this with an deleted_at column? Not sure, but something to think about.

Reply:
That's a good idea. I think eventually we can introduce an ORM and define default scopes that will exclude the soft deleted ones.